alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/SecureDestinationRestApiIT.kt

```diff
@@ -86,6 +86,130 @@ class SecureDestinationRestApiIT : AlertingRestTestCase() {
         }
     }
 
+    fun `test update destination with disable filter by`() {
+        disableFilterBy()
+
+        val chime = Chime("http://abc.com")
+        val destination = Destination(
+            type = DestinationType.CHIME,
+            name = "test",
+            user = randomUser(),
+            lastUpdateTime = Instant.now(),
+            chime = chime,
+            slack = null,
+            customWebhook = null,
+            email = null)
+        val createdDestination = createDestination(destination = destination)
+
+        assertEquals("Incorrect destination name", createdDestination.name, "test")
+        assertEquals("Incorrect destination type", createdDestination.type, DestinationType.CHIME)
+
+        val slack = Slack("http://abc.com")
+        val destinationV2 = createdDestination.copy(name = "testUpdate", type = DestinationType.SLACK, chime = null, slack = slack)
+
+        val updatedDestination = updateDestination(destination = destinationV2)
+        assertEquals("Incorrect destination name", updatedDestination.name, "testUpdate")
+        assertEquals("Incorrect destination type", updatedDestination.type, DestinationType.SLACK)
+    }
+
+    fun `test update destination with enable filter by`() {
+        enableFilterBy()
+        if (!isHttps()) {
+            // if security is disabled and filter by is enabled, we can't create monitor
+            // refer: `test create destination with enable filter by`
+            return
+        }
+
+        val chime = Chime("http://abc.com")
+        val destination = Destination(
+            type = DestinationType.CHIME,
+            name = "test",
+            user = randomUser(),
+            lastUpdateTime = Instant.now(),
+            chime = chime,
+            slack = null,
+            customWebhook = null,
+            email = null)
+
+        // 1. create a destination as admin user
+        val createdDestination = createDestination(destination, true)
+
+        assertEquals("Incorrect destination name", createdDestination.name, "test")
+        assertEquals("Incorrect destination type", createdDestination.type, DestinationType.CHIME)
+
+        val slack = Slack("http://abc.com")
+        val destinationV2 = createdDestination.copy(name = "testUpdate", type = DestinationType.SLACK, chime = null, slack = slack)
+
+        val updatedDestination = updateDestination(destination = destinationV2)
+        assertEquals("Incorrect destination name", updatedDestination.name, "testUpdate")
+        assertEquals("Incorrect destination type", updatedDestination.type, DestinationType.SLACK)
```

Comment:
I am wondering the use case for this filterBy feature: another user B with different backend role compared to user A who create the destination, when fillterBy enabled, user B will not be able to update the the destination, even user B has the permission for update destination API

Reply:
Yes, we will be adding those. Sriram modified the testing framework in this [PR](https://github.com/opendistro-for-elasticsearch/alerting/pull/313/), so it will be easy to add a different user to test this use case. I have tested this manually for now.